godot-core/src/builtin/collections/packed_array.rs

```diff
@@ -9,927 +9,756 @@
-            /// To obtain Rust slices, see [`as_slice`][Self::as_slice] and [`as_mut_slice`][Self::as_mut_slice].
-            #[doc(alias = "slice")]
-            // TODO(v0.3): change to i32 like NodePath::slice/subpath() and support+test negative indices.
-            pub fn subarray(&self, begin: usize, end: usize) -> Self {
-                let len = self.len();
-                let begin = begin.min(len);
-                let end = end.min(len);
-                self.as_inner().slice(to_i64(begin), to_i64(end))
-            }
+    /// Appends another array at the end of this array.
+    ///
+    /// _Godot equivalent: `append_array`_
+    pub fn extend_array(&mut self, other: &PackedArray<T>) {
+        // Rust only, to be benchmarked:  self.extend(other.as_slice().iter().cloned());
 
-            /// Returns a shared Rust slice of the array.
-            ///
-            /// The resulting slice can be further subdivided or converted into raw pointers.
-            ///
-            /// See also [`as_mut_slice`][Self::as_mut_slice] to get exclusive slices, and
-            /// [`subarray`][Self::subarray] to get a sub-array as a copy.
-            pub fn as_slice(&self) -> &[$Element] {
-                if self.is_empty() {
-                    &[]
-                } else {
-                    let data = self.ptr(0);
+        T::op_append_array(self.as_inner(), other);
+    }
 
-                    // SAFETY: PackedArray holds `len` elements in contiguous storage, all of which are initialized.
-                    // The array uses copy-on-write semantics, so the slice may be aliased, but copies will use a new allocation.
-                    unsafe {
-                        std::slice::from_raw_parts(data, self.len())
-                    }
-                }
-            }
+    /// Converts this array to a Rust vector, making a copy of its contents.
+    pub fn to_vec(&self) -> Vec<T> {
+        self.as_slice().to_vec()
+    }
 
-            /// Returns an exclusive Rust slice of the array.
-            ///
-            /// The resulting slice can be further subdivided or converted into raw pointers.
-            ///
-            /// See also [`as_slice`][Self::as_slice] to get shared slices, and
-            /// [`subarray`][Self::subarray] to get a sub-array as a copy.
-            pub fn as_mut_slice(&mut self) -> &mut [$Element] {
-                if self.is_empty() {
-                    &mut []
-                } else {
-                    let data = self.ptr_mut(0);
+    /// Returns a sub-range `begin..end`, as a new packed array.
+    ///
+    /// This method is called `slice()` in Godot.
+    /// The values of `begin` (inclusive) and `end` (exclusive) will be clamped to the array size.
+    ///
+    /// To obtain Rust slices, see [`as_slice`][Self::as_slice] and [`as_mut_slice`][Self::as_mut_slice].
+    #[doc(alias = "slice")]
+    // TODO(v0.3): change to i32 like NodePath::slice/subpath() and support+test negative indices.
```

Comment:
dead comment/leftover todo after copy-paste (and we still use usize instead of u32/i32 :smile: )

Reply:
Same one exists still in `Array` :grinning: so it's probably easier to do them all at once and not in this refactoring.
https://github.com/godot-rust/gdext/blob/bb37196d267c78863a3c268da1fed10d5980a56a/godot-core/src/builtin/collections/array.rs#L529
https://github.com/godot-rust/gdext/blob/bb37196d267c78863a3c268da1fed10d5980a56a/godot-core/src/builtin/collections/array.rs#L548